ci/tests/test_geneformer/test_geneformer_model.py

```diff
@@ -66,126 +66,86 @@ def test_ensembl_data_is_caught(self, geneformer, mock_data, gene_symbols, raise
+        config = GeneformerConfig(model_name="gf-12L-95M-i4096", batch_size=5, emb_mode=emb_mode)
+        geneformer = Geneformer(config)
+        mocker.patch.object(geneformer.model, "forward", return_value=mock_embeddings_v2)
+
+        dataset = geneformer.process_data(mock_data, gene_names ="gene_symbols")
+        embeddings = geneformer.get_embeddings(dataset)
+        if emb_mode == "gene":
+            data_list = pd.Series({
+                            "ENSG00000187583": np.array([1.0, 2.0, 3.0, 2.0, 1.0]),
+                            "ENSG00000187634": np.array([6.0, 6.0, 6.0, 6.0, 6.0]),
+                            "ENSG00000188290": np.array([5.0, 5.0, 5.0, 5.0, 5.0])
+                        })  
+            for key in data_list.index:
+                assert np.all(np.equal(embeddings[0][key], data_list[key]))
+
+        if emb_mode == "cls":
+            assert (embeddings == np.array([6.0, 5.0, 7.0, 5.0, 5.0])).all()
+        if emb_mode == "cell":  
+            expected = np.array([[4, 4.333333, 4.666667, 4.333333, 4]])
+            np.testing.assert_allclose(
+                embeddings, 
+                expected, 
+                rtol=1e-4,
+                atol=1e-4
+            )
+
+    @pytest.mark.parametrize("emb_mode", ["cell", "gene"])
+    def test_fine_tune_classifier_returns_correct_shape(self, emb_mode, mock_data):
+        fine_tuned_model = GeneformerFineTuningModel(GeneformerConfig(emb_mode=emb_mode), fine_tuning_head="classification", output_size=1)
+        tokenized_dataset = fine_tuned_model.process_data(mock_data, gene_names='gene_symbols')
+        tokenized_dataset = tokenized_dataset.add_column('labels', list([0]))
+        
+        fine_tuned_model.train(train_dataset=tokenized_dataset, label='labels')
 
-    def test_model_input_size(self, geneformer):
-        assert geneformer.config["input_size"] == geneformer.configurer.model_map[geneformer.config["model_name"]]['input_size']
+        outputs = fine_tuned_model.get_outputs(tokenized_dataset)
+        assert outputs.shape == (len(mock_data), 1)
 
-    def test_fine_tune_classifier_returns_correct_shape(self, mock_data, fine_tune_mock_data):
-        device = "cuda" if torch.cuda.is_available() else "cpu"
-        fine_tuned_model = GeneformerFineTuningModel(GeneformerConfig(device=device), fine_tuning_head="classification", output_size=1)
+    def test_fine_tune_classifier_cls_returns_correct_shape(self, mock_data):
+        fine_tuned_model = GeneformerFineTuningModel(GeneformerConfig(model_name="gf-12L-95M-i4096", emb_mode="cls"), fine_tuning_head="classification", output_size=1)
         tokenized_dataset = fine_tuned_model.process_data(mock_data, gene_names='gene_symbols')
-        tokenized_dataset = tokenized_dataset.add_column('labels', fine_tune_mock_data)
+        tokenized_dataset = tokenized_dataset.add_column('labels', list([0]))
```

Comment:
Is using list([0]) equivalent to just passing [0]? Could we remove the list() specification?

Reply:
Removed!